File: chk_zorome.go
//filename:chk_zorome.go

package main

import "strings"
import "fmt"
import "regexp"
import "flag"

func main () {
    flag.Parse()
    var s string = ""

    //引数が空だとsはからのまま。値があると代入。カンマ区切りだと要素１つ分のみ
    for i := 0; i < flag.NArg(); i++ {
        if i > 0 {
            s += " "
        }
        s += flag.Arg(i)
    }

    //スライスarr作成。カンマ区切りを要素に分解して代入
    var arr []string
    arr = strings.Split(s, ",")

    //要素ごとに判定
    //他言語で使うゾロ目の正規表現"^([0-9])\1{2}$"が使えなかったので10通り並べた
    for i := range arr {
        re_3zorome := regexp.MustCompile(`^(000|111|222|333|444|555|666|777|888|999)$`)
//        re_3zorome := regexp.MustCompile(`^([0-9])\\1{2}$`) //←うまくいかない・・・
        result1 := re_3zorome.MatchString(arr[i])
        if result1 == true {
            fmt.Println("zorome: ",arr[i])
        } else {
                re_3digits := regexp.MustCompile(`^[0-9]{3}$`)
                result2 := re_3digits.MatchString(arr[i])
                if result2 == true {
                    fmt.Println(arr[i])
                } else {
                        fmt.Println("unmatch pattern: ",arr[i])
                        }
                }
        }
}

File: chk_zorome.rb
###########################################################
#
# スクリプト名：chk_zorome.rb
# 概　　　　要：標準入力から受けたカンマ区切りの文字列、数値に応じて
#           　以下の処理を実施する。
#           ・3桁の数値ではない → unmach pattern: <受け取った値>
#           ・3桁のゾロ目の数値 → zorome: <受け取った値>
#           ・上記以外の数値   → <受け取った値>
#
############################################################

# 標準入力を受け付ける
print "カンマ区切りで文字列、数値を入力してください： "
ary_input = $stdin.gets.chomp.split(',')

# 入力された文字列または数値分ループをする
ary_input.each do |value|

  # 標準入力の値が3桁の数値か判定する
  if((/^[+-]?[0-9]{,3}?$/ =~ value.to_s) == nil) then
    print "unmach pattern: ", value, "\n"
    next
  end

  # 数値がゾロ目かどうか判定する
  if(/^[+-]?([1-9])\1+$/ =~ value.to_s) then
    print "zorome: ", value.to_i, "\n"
  else
    print value.to_i, "\n"
  end

end